packages/compiler/server/serverlib.ts

```diff
@@ -1121,57 +1120,94 @@ export function createServer(host: ServerHost): Server {
     return url ? host.getOpenDocumentByURL(url) : undefined;
   }
 
-  async function readFile(path: string): Promise<ServerSourceFile> {
-    // Try open files sent from client over LSP
-    const document = getOpenDocument(path);
-    if (document) {
-      return {
-        document,
-        ...createSourceFile(document.getText(), path),
-      };
-    }
+  async function fileURLToRealPath(url: string) {
+    return resolveRealPath(compilerHost, compilerHost.fileURLToPath(url));
+  }
+
+  function createFileSystemCache() {
+    const cache = new Map<string, CachedFile | CachedError>();
+    let changes: FileEvent[] = [];
+    return {
+      async get(path: string) {
+        for (const change of changes) {
+          const path = await fileURLToRealPath(change.uri);
```

Comment:
that one is a bit annoying to have to make the cache async because of this.

Reply:
Yes and it's kind of purpose defeating that we have to do some I/O to ask the file system for the real path in order to consult the cache whose purpose is to avoid repeated I/O.

Reply:
wonder if we could be caching the vscode path -> realpath results and when we invalidate the cache we don't need to check for realpath. 

I guess this always risk that vscode gives us a different path casing when invalidating than when we loaded first.

Reply:
or maybe the cache can be case insensitive when invalidating itself.

Reply:
fine keeping it like that too and see if we can improve as we go